lib/plug/parsers/multipart.ex

```diff
@@ -172,7 +172,21 @@ defmodule Plug.Parsers.MULTIPART do
         {:file, name, path, upload}
 
       :error ->
-        {:binary, name}
+        case Map.fetch(params, "filename*") do
+          {:ok, ""} ->
+            :skip
+
+          {:ok, "utf-8''" <> filename} ->
+            filename = URI.decode(filename)
```

Comment:
Should we check `String.valid?(filename)` after decoding?

Reply:
Or alternatively, raise on bad encoding: https://github.com/elixir-plug/plug/blob/d88a7e7aa52ab5d45596a1afac4d94ebae5f75bb/lib/plug/parsers/multipart.ex#L94

Reply:
After failing check with `String.valid?` we would probably need to return `:skip` here, which would be sort of silencing the problem, so I added call to `validate_utf8!`. Let me know if you'd like to tweak the context of raised error or anything else.